src/core/MOM_barotropic.F90

```diff
@@ -1609,6 +1628,36 @@ subroutine btstep(U_in, V_in, eta_in, dt, bc_accel_u, bc_accel_v, forces, pbce,
     call Filt_accum(vbt, vk1, CS%Time, US, CS%Filt_CS_vk1)
   endif
 
+  ! Apply frequency-dependent linear wave drag
+  Drag_u(:,:) = 0.0 ; Drag_v(:,:) = 0.0
+
+  if (CS%linear_freq_drag) then
+    !$OMP do
+    do j=js,je ; do I=is-1,ie
+      if (CS%lin_drag_um2(I,j) > 0.0 .or. CS%lin_drag_uk1(I,j) > 0.0) then
```

Comment:
It has been our experience that moving logical tests outside of do loops is almost always more efficient, even if it means duplicating a little code or (in this case) adding 0 to some expressions that did not need it.  Compliers would do optimizations like this if they have enough information, but a compiler would not know, for instance, that there should never be negative values of `lin_drag_um2` and `lin_drag_uk1`.

Reply:
Again I was following what has been done in the case of `CS%linear_wave_drag`. Here, it seems not possible to completely eliminate logical tests inside do loops, because the calculation should only be performed where `Htot` is non-zero. I did remove the two individual logical tests for `CS%lin_drag_um2(I,j) > 0.0` and `CS%lin_drag_uk1(I,j) > 0.0` in both loops though, and this should add some efficiency.